sound/soc/sof/intel/hda-dai.c

```diff
@@ -56,13 +63,22 @@ hda_dai_get_ops(struct snd_pcm_substream *substream, struct snd_soc_dai *cpu_dai
 	struct snd_soc_dapm_widget *w = snd_soc_dai_get_widget(cpu_dai, substream->stream);
 	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(cpu_dai->component);
 	struct snd_sof_widget *swidget = w->dobj.private;
-	struct snd_sof_dai *sdai = swidget->private;
+	const struct hda_dai_widget_dma_ops *ops;
+	struct snd_sof_dai *sdai;
+
+	if (!swidget) {
+		ops = hda_select_dai_widget_ops(sdev, NULL);
+		if (!ops || !ops->get_hext_stream || !ops->setup_hext_stream)
+			return NULL;
+
+		return ops;
+	}
+
+	sdai = swidget->private;
 
 	/* select and set the DAI widget ops if not set already */
 	if (!sdai->platform_private) {
-		const struct hda_dai_widget_dma_ops *ops =
-			hda_select_dai_widget_ops(sdev, swidget);
-
+		ops = hda_select_dai_widget_ops(sdev, swidget);
```

Comment:
I was not able to follow the change from the initial code written by @ranj063 

In one case sdai->platform_private is required and not in the other.

Reply:
there's no swidget and so there's no sdai with dspless mode @plbossart. so there's cant be a platform_private either

Reply:
This change might not be needed if the swidget allocation way works.